src/hint_processor/builtin_hint_processor/math_utils.rs

```diff
@@ -374,20 +374,17 @@ pub fn is_positive(
     ap_tracking: &ApTracking,
 ) -> Result<(), HintError> {
     let value = get_integer_from_var_name("value", vm, ids_data, ap_tracking)?;
+    let value_as_int = value.to_signed_felt();
     let range_check_builtin = vm.get_range_check_builtin()?;
     //Main logic (assert a is positive)
     match &range_check_builtin._bound {
-        Some(bound) if &value.abs() > bound => {
+        Some(bound) if value_as_int.abs() > bound.to_bigint() => {
```

Comment:
This is particularly costly. Can we check if we find an alternative solution? I'm not 100% sure I'm right in [this PR](https://github.com/lambdaclass/cairo-rs/pull/935/files) (which is why it's a draft), but maybe it can serve as inspiration?